http-api/src/v1/projects.rs

```diff
@@ -720,3 +720,402 @@ fn resolve_revisions(
         revisions,
     }
 }
+
+mod setup {
+    use std::path::Path;
+    use std::{env, fs};
+
+    use git2::Oid;
+
+    use librad::crypto::keystore::crypto::{Pwhash, KDF_PARAMS_TEST};
+    use librad::crypto::keystore::pinentry::SecUtf8;
+    use librad::crypto::BoxedSigner;
+    use librad::git::identities::Project;
+    use librad::git::util;
+    use librad::git_ext::tree;
+    use librad::profile::{Profile, LNK_HOME};
+
+    use radicle_common::cobs::patch::MergeTarget;
+    use radicle_common::cobs::shared::Store;
+    use radicle_common::{keys, person, profile, project, test};
+
+    #[allow(dead_code)]
+    pub fn profile() -> (Profile, BoxedSigner, Project, Oid) {
+        let tempdir = env::temp_dir().join("rad").join("home").join("api");
+        let home = env::var(LNK_HOME)
+            .map(|s| Path::new(&s).to_path_buf())
+            .unwrap_or_else(|_| tempdir.to_path_buf());
+
+        env::set_var(LNK_HOME, home);
+
+        let name = "cloudhead";
+        let pass = Pwhash::new(SecUtf8::from(test::USER_PASS), *KDF_PARAMS_TEST);
+        let (profile, peer_id) = profile::create(profile::home(), pass.clone()).unwrap();
+        let signer = test::signer(&profile, pass).unwrap();
+        let storage = keys::storage(&profile, signer.clone()).unwrap();
+        let person = person::create(&profile, name, signer.clone(), &storage).unwrap();
+
+        person::set_local(&storage, &person).unwrap();
+
+        let whoami = person::local(&storage).unwrap();
+        let payload = project::payload(
+            "nakamoto".to_owned(),
+            "Bitcoin light-client".to_owned(),
+            "master".to_owned(),
+        );
+        let project = project::create(payload, &storage).unwrap();
+
+        let commit = util::quick_commit(
+            &storage,
+            &project.urn(),
+            vec![
+                ("HI", tree::blob(b"Hi Bob")),
+                ("README", tree::blob(b"This is a readme")),
+            ]
+            .into_iter()
+            .collect(),
+            "say hi to bob",
+        )
+        .unwrap();
+
+        // create remote head
+        // otherwise testing `.../patches/:id` would fail
```

Comment:
we need to create and point `refs/namespaces/<urn>/refs/remotes/<remotes>/heads/master` to tip of repo otherwise `.../patches/:id` wouldn't work.

I did it using `fs::write`, as doing it using `git2` was not very straightforward.